### ISSUE TITLE
404 SXA Page not found link redirects instead of staying on original url

### ISSUE THREAD

**Issue by bewards** [Sitecore/jss#1416]
## Description
The standard SXA Page not found field functionality when loading a url that's non-existent will provide the end user with the 404 content, a 404 status code, and the original url (for SEO).

## Expected behavior
The Next.js middleware seems to do a redirect instead of a rewrite. Loading a non-existent page such as "/nonexistant" 307 redirects me to the 404 not found page and changes the url.

## Your Environment
* Sitecore Version: 10.3 XM1 Next.js
* JSS Version: 21.0.5
* Next.js Version: `"next": "^12.3.4"`
* Browser Name and version: Chrome Latest
* Operating System and version (desktop or mobile): Windows 11 desktop

## Screenshots
**original url redirected and changed url**
![404-not-found_2023-04-04](https://user-images.githubusercontent.com/3927169/229924726-d2b45256-5eb4-4ead-be2b-7e4bec64806c.png)

## rendering container log for request
```yaml
2023-04-05 10:23:27 2023-04-05T14:23:28.521Z sitecore-jss:layout fetching layout data for /notapage en healthplan
2023-04-05 10:23:27 2023-04-05T14:23:28.521Z sitecore-jss:layout request: { url: 'http://cm/sitecore/api/graph/edge', headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' }, query: 'query {\n' + '      layout(site:"healthplan", routePath:"/notapage", language:"en"){\n' + '        item {\n' + '          rendered\n' + '        }\n' + '      }\n' + '    }', variables: undefined }
2023-04-05 10:23:27 2023-04-05T14:23:28.545Z sitecore-jss:layout response: { layout: null }
2023-04-05 10:23:27 2023-04-05T14:23:28.546Z sitecore-jss:dictionary fetching dictionary data for en healthplan
2023-04-05 10:23:27 2023-04-05T14:23:28.546Z sitecore-jss:dictionary request: { url: 'http://cm/sitecore/api/graph/edge', headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' }, query: '\n' + '  query DictionarySearch(\n' + '    $rootItemId: String!\n' + '    $language: String!\n' + '    $templates: String!\n' + '    $pageSize: Int = 10\n' + '    $after: String\n' + '  ) {\n' + '    search(\n' + '      where: {\n' + '        AND: [\n' + '          { name: "_path", value: $rootItemId, operator: CONTAINS }\n' + '          { name: "_language", value: $language }\n' + '          { name: "_templates", value: $templates, operator: CONTAINS }\n' + '        ]\n' + '      }\n' + '      first: $pageSize\n' + '      after: $after\n' + '    ) {\n' + '      total\n' + '      pageInfo {\n' + '        endCursor\n' + '        hasNext\n' + '      }\n' + '      results {\n' + '        key: field(name: "Key") {\n' + '          value\n' + '        }\n' + '        phrase: field(name: "Phrase") {\n' + '          value\n' + '        }\n' + '      }\n' + '    }\n' + '  }\n', variables: { rootItemId: '{E4BF95F2-86D8-47C9-AF87-A85D0EFE2348}', language: 'en', templates: '6d1cd89719364a3aa511289a94c2a7b1', pageSize: undefined, after: '' } }
2023-04-05 10:23:27 2023-04-05T14:23:28.583Z sitecore-jss:dictionary response: { search: { total: 10, pageInfo: { endCursor: 'MTA=', hasNext: false }, results: [ [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object] ] } }
2023-04-05 10:23:27 2023-04-05T14:23:28.584Z sitecore-jss:errorpages request: { url: 'http://cm/sitecore/api/graph/edge', headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' }, query: '\n' + '  query ErrorPagesQuery($siteName: String!, $language: String!) {\n' + '    site {\n' + '      siteInfo(site: $siteName) {\n' + '        errorHandling(language: $language) {\n' + '          notFoundPagePath\n' + '          serverErrorPagePath\n' + '        }\n' + '      }\n' + '    }\n' + '  }\n', variables: { siteName: 'healthplan', language: 'en' } }
2023-04-05 10:23:27 2023-04-05T14:23:28.611Z sitecore-jss:errorpages response: { site: { siteInfo: { errorHandling: [Object] } } }
2023-04-05 10:23:28 sitecore-jss:redirects request: {
2023-04-05 10:23:28   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:28   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:28   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:28   variables: { siteName: 'healthplan' }
2023-04-05 10:23:28 } +2s
2023-04-05 10:23:28 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +14ms
2023-04-05 10:23:29 2023-04-05T14:23:30.107Z sitecore-jss:layout fetching layout data for /404-Not-Found en healthplan
2023-04-05 10:23:29 2023-04-05T14:23:30.107Z sitecore-jss:layout request: { url: 'http://cm/sitecore/api/graph/edge', headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' }, query: 'query {\n' + '      layout(site:"healthplan", routePath:"/404-Not-Found", language:"en"){\n' + '        item {\n' + '          rendered\n' + '        }\n' + '      }\n' + '    }', variables: undefined }
2023-04-05 10:23:29 2023-04-05T14:23:30.294Z sitecore-jss:layout response: { layout: { item: { rendered: [Object] } } }
2023-04-05 10:23:29 2023-04-05T14:23:30.294Z sitecore-jss:dictionary using cached dictionary data for en healthplan
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +2s
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +15ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +10ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +9ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +10ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +3ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +11ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +11ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +15ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +1ms
2023-04-05 10:23:29 sitecore-jss:redirects request: {
2023-04-05 10:23:29   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:29   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:29   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:29   variables: { siteName: 'healthplan' }
2023-04-05 10:23:29 } +15ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +7ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +2ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +5ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +3ms
2023-04-05 10:23:29 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +16ms
2023-04-05 10:23:30 sitecore-jss:redirects request: {
2023-04-05 10:23:30   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:30   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:30   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:30   variables: { siteName: 'healthplan' }
2023-04-05 10:23:30 } +990ms
2023-04-05 10:23:30 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +19ms
2023-04-05 10:23:31 sitecore-jss:redirects request: {
2023-04-05 10:23:31   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:31   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:31   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:31   variables: { siteName: 'healthplan' }
2023-04-05 10:23:31 } +251ms
2023-04-05 10:23:31 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +20ms
2023-04-05 10:23:31 sitecore-jss:redirects request: {
2023-04-05 10:23:31   url: 'http://cm/sitecore/api/graph/edge',
2023-04-05 10:23:31   headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' },
2023-04-05 10:23:31   query: '\n  query RedirectsQuery($siteName: String!) {\n    site {\n      siteInfo(site: $siteName) {\n        redirects {\n          pattern\n          target\n          redirectType\n          isQueryStringPreserved\n          locale\n        }\n      }\n    }\n  }\n',
2023-04-05 10:23:31   variables: { siteName: 'healthplan' }
2023-04-05 10:23:31 } +58ms
2023-04-05 10:23:31 sitecore-jss:redirects response: { site: { siteInfo: { redirects: [ length: 0 ] } } } +17ms
2023-04-05 10:23:32 2023-04-05T14:23:33.544Z sitecore-jss:layout fetching layout data for /404-Not-Found en healthplan
2023-04-05 10:23:32 2023-04-05T14:23:33.544Z sitecore-jss:layout request: { url: 'http://cm/sitecore/api/graph/edge', headers: { sc_apikey: '6FD03845-4AB4-47CE-9349-2B6FE898D68A' }, query: 'query {\n' + '      layout(site:"healthplan", routePath:"/404-Not-Found", language:"en"){\n' + '        item {\n' + '          rendered\n' + '        }\n' + '      }\n' + '    }', variables: undefined }
2023-04-05 10:23:32 2023-04-05T14:23:33.576Z sitecore-jss:layout response: { layout: { item: { rendered: [Object] } } }
2023-04-05 10:23:32 2023-04-05T14:23:33.576Z sitecore-jss:dictionary using cached dictionary data for en healthplan
```

**Comment by illiakovalenko** [comment]
@bewards Thank you for your ticket, I created a corresponding ticket in our inbox for further discussion

**Comment by illiakovalenko** [comment]
@bewards I close this issue, it was fixed here #1469
